### PR TITLE
Binance & Coinbase destination tag support

### PR DESCRIPTION
**Pull request by amjad1210**
#1816

### CODE REVIEW

**Comment by amjad1210** [comment]
@timmolter Any chance you can merge this soon?

**Comment by timmolter** [comment]
can you fix the errors please?
<https://travis-ci.org/knowm/XChange/jobs/560157055>

**Comment by amjad1210** [comment]
> can you fix the errors please?
> https://travis-ci.org/knowm/XChange/jobs/560157055

Fixed.

File: xchange-core/src/main/java/org/knowm/xchange/dto/account/DepositAddress.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package org.knowm.xchange.dto.account;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+public final class DepositAddress implements Serializable {
```
Comment (walec51) [inline xchange-core/src/main/java/org/knowm/xchange/dto/account/DepositAddress.java]: we already have a very generic interface WithdrawFundsParams which you can extend to add a tag 
we also already have this address in DefaultWithdrawFundsParams which already has the address
we also already have RippleWithdrawFundsParams that adds a tag for ripple and MoneroWithdrawFundsParams with add paymentId <- this is a little unfortunate as they are in essence the same this

please remove DepositAddress , add destinationTag to DefaultWithdrawFundsParams with the following documentation:

```
  /**
   * Various cryptocurrencies allow to add an additional identifier / tag to you transaction which helps them
   * to process you payment. Ripple for example has a tag, Monero has a payment id.
   */
  public String getDestinationTag() {
```
and us it instead in Binance and Coinbase

in another PR I'll then refactor the code to deprecate RippleWithdrawFundsParams and MoneroWithdrawFundsParams and add some better WithdrawFundsParams  interfaces

Comment (walec51) [inline xchange-core/src/main/java/org/knowm/xchange/dto/account/DepositAddress.java]: I would rename this to `AddressWithTag` as it is used both for deposits and withdrawals

File: xchange-core/src/main/java/org/knowm/xchange/dto/account/FundingRecord.java [hunk 2]
```diff
@@ -18,6 +18,9 @@
   /** Crypto currency address for deposit/withdrawal */
   private final String address;
 
+  /** Crypto currency destination tag for deposit/withdrawal */
+  private String destinationTag;
```
Comment (walec51) [inline xchange-core/src/main/java/org/knowm/xchange/dto/account/FundingRecord.java]: final

File: xchange-core/src/main/java/org/knowm/xchange/dto/account/DepositAddress.java (hunk 1 above)
Comment (amjad1210) [reply xchange-core/src/main/java/org/knowm/xchange/dto/account/DepositAddress.java]: Done.

File: xchange-core/src/main/java/org/knowm/xchange/dto/account/FundingRecord.java (hunk 2 above)
Comment (amjad1210) [reply xchange-core/src/main/java/org/knowm/xchange/dto/account/FundingRecord.java]: Fixed.

**Pull request merged**